src/snowflake/connector/auth/keypair.py

```diff
@@ -193,21 +190,24 @@ def handle_timeout(
         password: str | None,
         **kwargs: Any,
     ) -> None:
-        if self._retry_ctx.get_current_retry_count() > self._jwt_retry_attempts:
-            logger.debug("Exhausted max login attempts. Aborting connection")
-            self._retry_ctx.reset()
-            raise OperationalError(
-                msg=f"Could not connect to Snowflake backend after {self._retry_ctx.get_current_retry_count()} attempt(s)."
-                "Aborting",
-                errno=ER_FAILED_TO_CONNECT_TO_DB,
-            )
-        else:
-            logger.debug(
-                f"Hit JWT timeout, attempt {self._retry_ctx.get_current_retry_count()}. Retrying..."
-            )
-            self._retry_ctx.increment_retry()
+        logger.debug("Invoking base timeout handler")
+        super().handle_timeout(
+            authenticator=authenticator,
+            service_name=service_name,
+            account=account,
+            user=user,
+            password=password,
+            delete_params=False,
+        )
+
+        logger.debug("Base timeout handler passed, checking for JWT expiry")
+        jwt_timed_out = False
+        if self._retry_ctx.remaining_time_millis(jwt_timed_out) is not None:
+            jwt_timed_out = self._retry_ctx.remaining_time_millis(jwt_timed_out) < 0
```

Comment:
you can do it in one line:
```suggestion
        jwt_timed_out = False if self._retry_ctx.remaining_time_millis(jwt_timed_out) is None else self._retry_ctx.remaining_time_millis(jwt_timed_out) < 0
```